app/misc/graphing.ts

```diff
@@ -583,11 +525,70 @@ function makeAbsNode(c, column: number) {
             count: 1,
         });
     }
+
+    // link the branch to updated absNode
+    if (c.toString() in bname) {
+        for (let i = 0; i < bname[c.toString()].length; i++) {
+            let branchName = bname[c.toString()][i];
+            let bp = branchName.name().split("/");
+            let shortName = bp[bp.length - 1];
+            console.log(shortName + " sub-branch: " + branchName.isHead().toString());
+            if (branchName.isHead()) {
+                shortName = "*" + shortName;
+            }
+            let bsnodeId = generateUniqueNumber();
+            abNodes.add({
+                id: bsnodeId,
+                shape: "box",
+                title: branchName,
+                label: shortName,
+                physics: false,
+                fixed: false,
+                x: (column - 0.6 * (i + 1)) * spacingX,
+                y: (nodeId - 0.3) * spacingY,
+                nodeType: NodeType.Branch
+            });
+
+            abEdges.add({
+                from: bsnodeId,
+                to: nodeId
+            });
+        }
+    }
+
+    // link the tag to updated absNode
+    if (c.toString() in tags) {
+        for (let i = 0; i < tags[c.toString()].length; i++) {
+            let tagName = tags[c.toString()][i];
+            let tp = tagName.name().split("/");
```

Comment:
The only place `tp` gets used is in the line below, I think it is just a stylistic choice but I would recommend just putting the variable definition into one line.